bot/cogs/geo_cog/streetviewrandomizer/street_view_random.py

```diff
@@ -0,0 +1,249 @@
+        gdf = gpd.read_file(SHAPE_FILE)
+        country = args['countries'][0]
+        new_country_selections_if_error = list()
+
+        gdf = gdf.loc[gdf['ISO3'] == f"{country}"]
+        gdf = gdf.assign(IMAGES=0)
+
+        total_attempts = 0
+        total_elapsed_time_ms = 0
+
+        # Loop for the amount of samples
+        for i in range(args['samples']):
+            fai = await self.find_available_image(gdf, args['radius'])
+            loops: int = 0
+            while fai.error_code == ERROR:
+                if loops >= 1:
+                    country = "SGP"
+                    ro = 25
+                else:
+                    loop = self.generate_country_options()[0]
+                    while loop in args['countries']:
+                        loop = self.generate_country_options()[0]
+
+                    country = loop
+                    ro = COUNTRIES[country]
+
+                gdf = gpd.read_file(SHAPE_FILE)
+                gdf = gdf.loc[gdf['ISO3'] == f"{country}"]
+                gdf = gdf.assign(IMAGES=0)
+                fai = await self.find_available_image(gdf, ro)
+                loops += 1
+
+            coord = fai.coord
+            country_df = fai.country_df
+            attempts = fai.attempts
+            elapsed_time_ms = fai.total_elapsed_time_ms
+            total_attempts += attempts
+            total_elapsed_time_ms += elapsed_time_ms
+
+            country_iso3 = country_df["ISO3"].values[0]
+            country_name = country_df["NAME"].values[0]
+            gdf.loc[gdf["ISO3"] == country_iso3, "IMAGES"] += 1
+
+            logging.log(1, f"\n> Image found in {country_iso3} ({country_name}) | "
+                           f"lon: {coord.lon}, lat: {coord.lat} | attempts: {attempts} "
+                           f"| total elapsed time: {elapsed_time_ms / 1000:.2f}s\n")
+            total_elapsed_time_ms = total_elapsed_time_ms + int(await self.save_images(
+                coord, args['size'], args['headings'], args['pitches'], args['fovs']
+            ))
+            logging.log(1, f"\nImage saved!\n Total attempts: {total_attempts} Average number of attempts per "
+                           f"sampling: {total_attempts / args['samples']:.2f} "
+                           f"\nTotal elapsed time: {total_elapsed_time_ms / 1000:.2f}s Average elapsed time per "
+                           f"sampling: "
+                           f"{(total_elapsed_time_ms / 1000) / args['samples']:.2f}s")
+
+        size = args['size']
+
+        return (('https://maps.googleapis.com/maps/api/streetview?'
+                + f'size={size}&'
+                + f'location={coord.lat},{coord.lon}&'
+                + f"heading={args['headings']}&"
+                + f"pitch={args['pitches']}&"
+                + f"fov={args['fovs']}&"
+                + f"key={bot_secrets.secrets.geocode_key}"),
+                coord.lat,
+                coord.lon,
+                country_iso3,
```

Comment:
If `args['samples']` is empty, `country_iso3` could be undefined and could raise an exception. Define it before the for-loop.